geom/type_line_string.go

```diff
@@ -16,32 +17,26 @@ import (
 //
 // 1. It must contain at least 2 distinct points.
 type LineString struct {
-	lines []Line
+	segments []int // index into coords
+	coords   []Coordinates
```

Comment:
Right so we wish to allow duplicated coords, but the segments will be the minimal set of lines. (I was wondering why you didn't just reduce the coords to the minimal set and deduce the lines rather than store a separate segments array, but I think that's by design.)

Reply:
Yep, to match the behaviour of PostGIS, we need to retain duplicated coordinates. But we also want to ignore duplicated coordinates when we traverse the line segments that make up the linestring (because we've defined Lines to have non-zero length, and lots of parts in the code make use of that definition).

Another way I could have done this was to reduce the coords to the minimal set (like we had before), and then store the indexes of _duplicated_ coordinates in another slice. I didn't try out that approach -- the advantage of doing that is that the `LineN` accessor wouldn't have to perform the extra level of redirection. The downside is that the `PointN` operations would no longer be constant time (they would be logarithmic instead). I'm going to try out that approach and see how I like it.